src/main/scala/it/pagopa/interop/authorizationprocess/common/AuthorizationUtils.scala

```diff
@@ -0,0 +1,35 @@
+package it.pagopa.interop.authorizationprocess.common
+
+import cats.implicits._
+import it.pagopa.interop.commons.utils.AkkaUtils.getOrganizationIdFutureUUID
+import it.pagopa.interop.authorizationprocess.error.AuthorizationProcessErrors._
+import java.util.UUID
+import scala.concurrent.{ExecutionContext, Future}
+import it.pagopa.interop.authorizationprocess.service.AuthorizationManagementService
+
+object AuthorizationUtils {
+
+  def assertIsClientConsumer(clientId: UUID)(
+    authorizationManagementService: AuthorizationManagementService
+  )(implicit contexts: Seq[(String, String)], ec: ExecutionContext): Future[Unit] =
+    for {
+      organizationId <- getOrganizationIdFutureUUID(contexts)
+      _              <- authorizationManagementService
+        .getClient(clientId)(contexts)
+        .ensureOr(client => OrganizationNotAllowedOnClient(clientId.toString, client.consumerId))(
+          _.consumerId == organizationId
+        )
+    } yield ()
+
+  def assertIsPurposeConsumer(purposeId: UUID, consumerId: UUID)(implicit
+    contexts: Seq[(String, String)],
+    ec: ExecutionContext
+  ): Future[Unit] =
+    for {
+      organizationId <- getOrganizationIdFutureUUID(contexts)
+      _              <- Future
+        .failed(OrganizationNotAllowedOnPurpose(purposeId.toString, consumerId.toString))
+        .unlessA(consumerId == organizationId)
+
+    } yield ()
+}
```

Comment:
I suggested this change to avoid code duplication, because now the `assertIsConsumer` is used in multiple services.
I may agree that passing the service is not ideal, but I'm not sure if I would prefer to have the entire method copied, future changes could easily lead to bugs.

@TonioGela do you have any suggestion for a solution that avoids code duplication?

Reply:
We can maybe reduce the attack surface of the method and keep it here. 
I would obtain the `organizationId` preemptively outside the method, so we won't need an `ExecutionContext` to be passed as well.I will then pass directly the `client` item and use a logic similar to the `assertIsPurposeConsumer`. My fear is that passing the whole `AuthorizationManagementService` lets the method perform anything towards that service.

My idea is more or less something like this (not sure if compiles, I wrote in in the PR page) whose scope is immensely limited. 

```scala
  def assertIsClientConsumer(organizationId: String)(client: Client): Future[Unit] =
     Future.failed(OrganizationNotAllowedOnClient(client.id.toString, client.consumerId))
       .unlessA(client.consumerId == organizationId)
```

Reply:
It is definitely cleaner, but we would still have most of the code duplicated.
Also, moving the method to the original class would hide the `ExecutionContext` from the signature, but it would be used anyway, I don't see practical gain in this case.

If you have a strong opinions on this we could split the method as you suggested, and in each class create duplicated methods to retrieve the `client`.
I would just keep the extraction of the `organizationId` here, we can get rid off the `ExecutionContext` by using `Either` instead of `Future` (and the `getOrganizationIdUUID` helper).

What do you think?

Reply:
> I would just keep the extraction of the `organizationId` here, we can get rid off the `ExecutionContext` by using `Either` instead of `Future` (and the `getOrganizationIdUUID` helper).

That's an awesome trade-off, good idea. It's not clear to me if you are suggesting to keep the `AuthorizationManagementService` in the signature as well though. (I suppose not since `Either`)

Reply:
Correct, I was thinking of keeping these functions here, without the `authorizationManagementService.getClient` call, and create two identical functions in the two services, which would perform the `getClient` and call these functions

Reply:
@nttdata-rtorsoli if it's not clear talk with @galales about this idea 👆